cirq/ops/linear_combinations.py

```diff
@@ -315,6 +316,71 @@ def copy(self) -> 'PauliSum':
         factory = type(self)
         return factory(self._linear_dict.copy())
 
+    def expectation(self,
+                    state: np.ndarray,
+                    qubit_map: Optional[Dict[raw_types.Qid, int]]=None
+                    ) -> float:
+        """Evaluate the expectation value of this PauliSum.
+
+        Compute the expectation value of this Hermitian PauliSum with respect to
+        an array representing either a wavefunction or density matrix.
+
+        See `PauliString.expectation` for input requirements.
+
+        Args:
+            state: An array representing a wavefunction or density matrix.
+            qubit_map: A map from all qubits used in this PauliSum to the
+            indices of the qubits that `state` is defined over.
+
+        Returns:
+            The expectation value of the input state.
+
+        Raises:
+            NotImplementedError if any term in this PauliSum is non-Hermitian.
+            ValueError if the shape of `state` is neither `(2 ** n,)`
+                or `(2 ** n, 2 ** n)`.
+        """
+        if any([abs(p.coefficient.imag) > 0.0001 for p in self]):
+            raise NotImplementedError(
+                "Cannot compute expectation value of a non-Hermitian "
+                "PauliString <{}>. Coefficient must be real.".format(self))
+
+        if qubit_map is None:
+            qubit_map = {q: i for i, q in enumerate(self.qubits)}
+        else:
+            if not isinstance(qubit_map, dict) or not all(
+                isinstance(k, raw_types.Qid) and isinstance(v, int) for k, v in qubit_map.items()):
+                raise TypeError("Input qubit map must be a valid mapping from "
+                                "Qubit ID's to integer indices.")
+
+        # `state` must support a map over all qubits in this PauliSum.
+        size = state.size
+        if state.shape == (size,):
+            num_qubits = size.bit_length() - 1
+            if len(set(qubit_map.keys())) > num_qubits:
```

Comment:
Having additional qubits in the state should be allowed. Expectation operators don't have to cover the entire state.